docs/specs/ISA.md

```diff
@@ -354,19 +354,20 @@ We have the following special opcodes tailored to optimize FRI proof verificatio
-| FRI_REDUCED_OPENING                                                                                                                                                                                                      | `a,b,c,4,e,f`   | Let `length = [e]_4`, `a_ptr = [a]_4`, `b_ptr = [b]_4`, `alpha = [f:EXT_DEG]_4`. `a_ptr` is the address of Felt array `a_arr` and `b_ptr` is the address of Ext array `b_arr`. Compute `sum((b_arr[i] - a_arr[i]) * alpha ^ i)` for `i=0..length` and write the value into `[c:EXT_DEG]_4`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
+| FRI_REDUCED_OPENING                                                                                                                                                                                                      | `a,b,c,d,e,f,g` | Let `a_ptr = [a]_4`, `b_ptr = [b]_4`, `length = [c]_4`, `alpha = [d:EXT_DEG]_4`, `hint_id = [f]_4`, `is_init = [g]_4`. `a_ptr` is the address of Felt array `a_arr` and `b_ptr` is the address of Ext array `b_arr`. Compute `sum((b_arr[i] - a_arr[i]) * alpha ^ i)` for `i=0..length` and write the value into `[e:EXT_DEG]_4`. It is required that `is_init` is boolean. If `is_init == 0`, read content of `a_arr` from the hint space at index `hint_id` and write into `a_arr`. Otherwise, read `a_arr` from memory.                                                                                                                                                                                                                                                                                                                                                                       |
```

Comment:
I feel like `is_init` is backwards? I read `is_init = 0` as `is_init = false`, in which case we should read from memory?

Reply:
No.. `is_init = false` means uninit => we should write to memory.

Reply:
Ah, I read at as, "is initial", not "is initialized".